src/functional/ffront/foast_passes/closure_var_folding.py

```diff
@@ -0,0 +1,46 @@
+# GT4Py Project - GridTools Framework
+#
+# Copyright (c) 2014-2022, ETH Zurich
+# All rights reserved.
+#
+# This file is part of the GT4Py project and the GridTools framework.
+# GT4Py is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the
+# Free Software Foundation, either vervisit_Constantsion 3 of the License, or any later
+# version. See the LICENSE.txt file at the top-level directory of this
+# distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+from typing import Any
+
+import functional.ffront.field_operator_ast as foast
+from eve import NodeTranslator, traits
+from eve.utils import ConstantNamespace
+
+
+class ClosureVarFolding(traits.VisitorWithSymbolTableTrait, NodeTranslator):
+    closure_vars: dict[str, Any]
+
+    def __init__(self, closure_vars: dict[str, Any]):
+        self.closure_vars = closure_vars
+
+    @classmethod
+    def apply(cls, node: foast.FieldOperator, closure_vars: dict[str, Any]):
+        return cls(closure_vars=closure_vars).visit(node)
+
+    def visit_Name(self, node: foast.Name, **kwargs):
+        if node.id in self.closure_vars:
```

Comment:
As discussed this is "vulnerable" to shadowing.

Reply:
Yeah, made a not to myself too. I tried to fix this but I had this issue that `kwargs["symtable]"` was existing but empty inside `visit_Name`. Do you know why that is, am I missing something?

Reply:
Fixed. Can you check it, the fix is not trivial?